### PR TITLE
Format numbers using number specifier

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Existing code was using `%v` to format numeric values. While that `%v` is a completely valid parameter, it could result in a recursive looping when a `fmt.Stringer` interface is being implemented on the argument.

Using `%d` eliminate this potential issue.

### CODE REVIEW

File: agreement/pseudonode.go [hunk 1]
```diff
@@ -484,7 +484,7 @@ func (t pseudonodeProposalsTask) execute(verifier *AsyncVoteVerifier, quit chan
 			Period:  uint64(vote.R.Period),
 		})
 	}
-	t.node.log.Infof("pseudonode.makeProposals: %v proposals created for round %v, period %v", len(verifiedVotes), t.round, t.period)
+	t.node.log.Infof("pseudonode.makeProposals: %v proposals created for round %d, period %d", len(verifiedVotes), t.round, t.period)
```
Comment (derbear) [inline agreement/pseudonode.go]: The first `%v` here should be changed too, right?

File: agreement/selector.go [hunk 2]
```diff
@@ -64,19 +64,19 @@ func membership(l LedgerReader, addr basics.Address, r basics.Round, p period, s
 
 	record, err := l.BalanceRecord(balanceRound, addr)
 	if err != nil {
-		err = fmt.Errorf("Service.initializeVote (r=%v): Failed to obtain balance record for address %v in round %v: %v", r, addr, balanceRound, err)
+		err = fmt.Errorf("Service.initializeVote (r=%v): Failed to obtain balance record for address %v in round %d: %v", r, addr, balanceRound, err)
```
Comment (derbear) [inline agreement/selector.go]: The first %v here (and below) should be changed too, right?

File: agreement/pseudonode.go (hunk 1 above)
Comment (tsachiherman) [reply agreement/pseudonode.go]: yes.

File: agreement/selector.go (hunk 2 above)
Comment (tsachiherman) [reply agreement/selector.go]: right.

**Pull request merged**